core/src/main/scala/org/apache/spark/shuffle/ShuffleMemoryManager.scala

```diff
@@ -27,6 +29,11 @@ import org.apache.spark.memory.{StaticMemoryManager, MemoryManager}
 import org.apache.spark.storage.{BlockId, BlockStatus}
 import org.apache.spark.unsafe.array.ByteArrayMethods
 
+abstract class MemoryConsumer {
+  @throws[IOException]
+  def release(numBytes: Long): Long
```

Comment:
We should add Scaladoc here to clarify that this is asking the consumer to release memory but that the consumer might release more or less memory than we asked for (or none at all!). Should we give this method a different name that suggests this, such as `askToRelease()`?